designs/CentralNuGetManagement/CPVM-Project_Package_Version_Override.md

```diff
@@ -0,0 +1,425 @@
+2. Provide a means for organizing multiple package versions together.
+3. Manage the package versions of transitive dependencies alongside direct dependencies.
+4. Follow the MSBuild approach with minimal changes to existing XML patterns.
+
+# Feature Behavior
+
+- All `<PackageVersion>` should be defined within `Directory.Packages.props`, and `Version` may not be defined on `<PackageReference>` within projects.
+    - Specifying a `Version` on a `<PackageReference>` will throw a build error.
+    - Specifying a `<PackageVersion>` within a project file is an anti-pattern, and it will generate a build time warning. It will not throw an error, and a developer may choose to suppress the warning by specifying `<AllowProjectPackageVersion>true</AllowProjectPackageVersion>` within a `<PropertyGroup>`.
+- A `<PackageVersion>` may have a `CentralManagementGroup` defined which allows a `<PackageReference>` to refer to this version using the `CentralManagementGroup` attribute.
+    - When a `Version` to be imported by a `<PackageReference>` needs to be different than the global group defined in `Directory.Packages.props`, a `CentralManagementGroup` can be specified on both the `<PackageReference>` in a project file and a corresponding `<PackageVersion>` in `Directory.Packages.props` that specifies the needed `Version`.
+    - A `<PackageReference>` may only have one `CentralManagementGroup` defined.
+    - A project can include references to multiple `CentralManagementGroup`.
+    - When `CentralManagementGroup` is not specified, both `<PackageVersion>` and `<PackageReference>` reference a global `CentralManagementGroup` that is automatically created.
+- Transitive dependencies are resolved based on finding a common level of specificity between direct imports.
+    - If a transitive dependency is imported by a package defined within the global group, then the `<PackageVersion>` for the transitive dependency from the global group applies.
+    - If a transitive dependency is imported by a package in multiple `CentralManagementGroup` that both include a `<PackageVersion>` for that transitive dependency, then the `<PackageVersion>` for the transitive dependency from the global group applies.
+    - If a transitive dependency is imported by a `<PackageReference>` with a `CentralManagementGroup` specified, then the `<PackageVersion>` for the transitive dependency from the `CentralManagementGroup` applies. If the transitive dependency does not have a matching `<PackageVersion>` within the same `CentralManagementGroup`, then the global group applies.
+    - If a transitive dependency is initiated by multiple packages within multiple, different `CentralManagementGroup` but **only one** of the `CentralManagementGroup` specifies a `<PackageVersion>` for the transitive dependency, then that version is used.
```

Comment:
Related with one of the above comments for supporting multiple groups inside of a project - in addition to the fact that this adds a great deal of complexity, the data for all the groups that a project needs need to carried to the NuGet core algorithms to  resolve the transitive dependencies. This could have performance impact.